lib/reverse_sentence.rb

```diff
@@ -1,6 +1,33 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n ** c), the time complexity is cuadratic because there is a nested loop which iterations depend on the size of the string
+# Space complexity: O(n), where n is the size of the array that will be created to store the pieces of the new string
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  i = 0
+  each_word_array = []
+  word = ""
+  reverse_sentence = ""
+  until i > my_sentence.length
+    if my_sentence[i] != " " && my_sentence[i] != nil
+      word += my_sentence[i]
+      i += 1
+    elsif my_sentence[i] == nil
+      each_word_array << word
+
+      (each_word_array.length - 1).times do |j|
+        reverse_sentence += "#{each_word_array[each_word_array.length - (j + 1)]}"
```

Comment:
I'm having a lot of trouble following this logic. This would be an excellent place for some elucidating comments. You could also consider breaking some of this functionality into a separate method in order to give it a name that hints at its purpose.